lib/rex/proto/kerberos/credential_cache/krb5_ccache_presenter.rb

```diff
@@ -0,0 +1,260 @@
+      output << 'Group IDs:'.indent(2)
+      output << validation_info.group_memberships.map { |group| "Relative ID: #{group.relative_id}, Attributes: #{group.attributes}".indent(4) }
+
+      output << "Logon Domain ID: #{validation_info.logon_domain_id}".indent(2)
+
+      output << "Effective Name: #{present_rpc_unicode_string(validation_info.effective_name)}".indent(2) # unless effective_name_info.to_s.blank?
+      output << "Full Name: #{present_rpc_unicode_string(validation_info.full_name)}".indent(2) # unless full_name_info.to_s.blank?
+      output << "Logon Script: #{present_rpc_unicode_string(validation_info.logon_script)}".indent(2) # unless logon_script_info.to_s.blank?
+      output << "Profile Path: #{present_rpc_unicode_string(validation_info.profile_path)}".indent(2) # unless profile_path_info.to_s.blank?
+      output << "Home Directory: #{present_rpc_unicode_string(validation_info.home_directory)}".indent(2) # unless home_directory_info.to_s.blank?
+      output << "Home Directory Drive: #{present_rpc_unicode_string(validation_info.home_directory_drive)}".indent(2) # unless home_directory_drive_info.to_s.blank?
+      output << "Logon Server: #{present_rpc_unicode_string(validation_info.logon_server)}".indent(2) # unless logon_server_info.to_s.blank?
+      output << "Logon Domain Name: #{present_rpc_unicode_string(validation_info.logon_domain_name)}".indent(2) # unless logon_domain_name_info.to_s.blank?
+
+      output.join("\n")
+    end
+
+    # @param [Object] info_buffer
+    def present_client_info(info_buffer)
+      client_info = info_buffer.buffer.pac_element
+      output = []
+      output << 'Client Info:'
+      output << "Name: '#{client_info.name.encode('utf-8')}'".indent(2)
+      output << "Client ID: #{present_time(client_info.client_id)}".indent(2)
+      output.join("\n")
+    end
+
+    # @param [Object] info_buffer
+    def present_server_checksum(info_buffer)
+      server_checksum = info_buffer.buffer.pac_element
+
+      sig = server_checksum.signature.bytes.map { |x| "\\x#{x.to_s(16).rjust(2, '0')}" }.join
+      "Pac Server Checksum:\n" +
+        "Signature: #{sig}".indent(2)
+    end
+
+    # @param [Object] info_buffer
+    def present_priv_server_checksum(info_buffer)
+      priv_server_checksum = info_buffer.buffer.pac_element
+
+      sig = priv_server_checksum.signature.bytes.map { |x| "\\x#{x.to_s(16).rjust(2, '0')}" }.join
+      "Pac Private Server Checksum:\n" +
+        "Signature: #{sig}".indent(2)
+    end
+
+    # @param [Object] info_buffer
+    def present_pac_info_buffer(info_buffer)
+      case info_buffer.ul_type
+      when 0x00000001
+        present_logon_info(info_buffer)
+      when 0x0000000A
+        present_client_info(info_buffer)
+      when 0x00000006
+        present_server_checksum(info_buffer)
+      when 0x00000007
+        present_priv_server_checksum(info_buffer)
+      else
+        info_buffer.to_s
+      end
+    end
+
```

Comment:
Placeholder comment to add the yard docs / replace `[Object]` with the appropriate types :+1: